src/common/cursor.c

```diff
@@ -295,25 +209,19 @@ void guac_common_cursor_set_argb(guac_common_cursor* cursor, int hx, int hy,
     cursor->hotspot_x = hx;
     cursor->hotspot_y = hy;
 
-    /* Update location based on new hotspot */
-    guac_protocol_send_move(cursor->client->socket, cursor->layer,
-            GUAC_DEFAULT_LAYER,
-            cursor->x - hx,
-            cursor->y - hy,
-            INT_MAX);
-
     /* Broadcast new cursor image to all users */
-    guac_protocol_send_size(cursor->client->socket, cursor->layer,
+    guac_protocol_send_size(cursor->client->socket, cursor->buffer,
             width, height);
 
-    guac_client_foreach_user(cursor->client, __send_user_cursor_image, cursor);
+    guac_client_stream_png(cursor->client, cursor->client->socket,
```

Comment:
It was actually never required.

I think this was written prior to implementation of `guac_client_stream_png()` and client-scoped streams. The old `guac_client_foreach_user()` call here just invoked `guac_user_stream_png()` for each connected user. That's equivalent to `guac_client_stream_png()` in that each user receives the image, but bad because the encoding process occurs once per user.

With `guac_client_stream_png()`, a stream is allocated at the client scope and the image (encoded only the once) is streamed over the broadcast socket.